draft-ietf-quic-http.md

```diff
@@ -376,6 +376,31 @@ as a stream error of type HTTP_CONNECT_ERROR ({{http-error-codes}}).
 Correspondingly, a proxy MUST send a TCP segment with the RST bit set if it
 detects an error with the stream or the QUIC connection.
 
+### Request Cancellation
+
+Either client or server can cancel requests by closing the stream (QUIC
+RST_STREAM or STOP_SENDING frames, as appropriate) with an error type of
+HTTP_REQUEST_CANCELLED ({{http-error-codes}}).  When the client cancels a
+request or response, it indicates that the response is no longer of interest.
+
+When the server cancels either direction of the request stream, it indicates
+that no application processing was performed.  The client can treat requests
```

Comment:
I don't think that this can be true.  It might indicate that no processing was performed, but we can't prohibit a server from cancelling streams after it does something, and this implies that requirement.

(This WAS true, because a variation on this text was used in the context of GOAWAY.)

Reply:
I think the clarification needed here is that I've defined "cancel" := "RST_STREAM with CANCELLED".  The server can, of course, RST_STREAM with some other error code at any time.  We can, I think, prohibit the server from using a particular error code after it does something.

Reply:
(Of course, we still have to spec the client behavior if the server fails in its obligation, which is what the subsequent text is about.)